aem-authoring-toolkit-plugin/src/main/java/com/exadel/aem/toolkit/core/util/writer/ComponentFacade.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exadel.aem.toolkit.core.util.writer;
+
+import com.exadel.aem.toolkit.api.annotations.main.Component;
+import com.exadel.aem.toolkit.core.exceptions.InvalidSettingException;
+import com.exadel.aem.toolkit.core.maven.PluginRuntime;
+
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Class that encapsulates steps taken by MVC pattern
```

Comment:
To be honest, a more thoroght explanation is desirable, because not clear "what steps" and "for what reason" etc. Also, javadoc is not for every method here